### PR TITLE
test: 💍 Add tests for SQFormAutocomplete

### PR DESCRIPTION
**Pull request by laurelbean**
I added a set of tests for SQFormAutocomplete.
I think I covered all the test cases, but please let me know if you think of another one.

✅ Closes: #229

### CODE REVIEW

**Comment by laurelbean** [comment]
I went ahead and added the tests for SQFormAsyncAutocomplete as well since they are very similar.

File: stories/__tests__/SQFormAutocomplete.stories.test.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      expect(label).toBeInTheDocument();
+      expect(label).toHaveAttribute('id', 'autocomplete-label');
+
+      const dropdown = screen.getByLabelText(/autocomplete/i);
+      expect(dropdown).toBeInTheDocument();
+
+      const openButton = screen.getByRole('button', {name: /open/i});
+      expect(openButton).toBeInTheDocument();
+
+      const submitButton = screen.getByRole('button', {
```
Comment (codyarose) [inline stories/__tests__/SQFormAutocomplete.stories.test.js:27]: Same thing here about removing the check for the submit button

**Review by codyarose — CHANGES_REQUESTED** [review]

File: stories/__tests__/SQFormAsyncAutocomplete.stories.test.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    it('should render with empty initial value', () => {
+      render(<SQFormAsyncAutocomplete size="auto" />);
+
+      const textField = screen.getByRole('textbox', {
+        name: /async autocomplete/i
+      });
+
+      expect(textField).toHaveTextContent('');
+    });
+
```
[the remark this answers is not in the document]
Comment (laurelbean) [reply stories/__tests__/SQFormAsyncAutocomplete.stories.test.js:42]: I did try and add that, but SQFormAsyncAutocomplete doesn't pass on the initial value. I'll file an issue for that.

[the remark this answers is not in the document]
Comment (codyarose) [reply stories/__tests__/SQFormAsyncAutocomplete.stories.test.js:37]: Merged #237 👍

File: stories/__tests__/SQFormAsyncAutocomplete.stories.test.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import * as stories from '../SQFormAsyncAutocomplete.stories';
+
+const {
+  Default: SQFormAsyncAutocomplete,
+  WithValidation: SQFormAsyncAutocompleteWithValidation
+} = composeStories(stories);
+
+describe('SQFormAsyncAutocomplete Tests', () => {
+  describe('AsyncAutocomplete Only', () => {
+    it('should render a dropdown, title, and 2 buttons', () => {
```
Comment (20BBrown14) [inline stories/__tests__/SQFormAsyncAutocomplete.stories.test.js:14]: This test description mentions two buttons but I'm only seeing one button being looked for in the test.

File: stories/__tests__/SQFormAutocomplete.stories.test.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import * as stories from '../SQFormAutocomplete.stories';
+
+const {
+  Default: SQFormAutocomplete,
+  WithValidation: SQFormAutocompleteWithValidation
+} = composeStories(stories);
+
+describe('SQFormAutocomplete Tests', () => {
+  describe('Autocomplete Only', () => {
+    it('should render a dropdown, title, and 2 buttons', () => {
```
Comment (20BBrown14) [inline stories/__tests__/SQFormAutocomplete.stories.test.js:14]: Same thing here about a mention of two buttons but only testing for one

Comment (20BBrown14) [inline stories/__tests__/SQFormAutocomplete.stories.test.js:14]: Same thing here about a mention of two buttons but only testing for one

File: stories/__tests__/SQFormAsyncAutocomplete.stories.test.js (hunk 3 above)
Comment (laurelbean) [reply stories/__tests__/SQFormAsyncAutocomplete.stories.test.js:14]: Good catch! I removed one of them in my last update, but didn't update the test name.

**Review by SeanGroff — APPROVED** [review]

**Pull request merged**